Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from ocpapi import find_adsorbate_binding_sites

results = await find_adsorbate_binding_sites(
    adsorbate="*OH",
    bulk="mp-126",
    adslab_filter=keep_slabs_with_miller_indices([(1, 1, 1)]),
)

ase_atoms = results.slabs[0].configs[0].to_ase_atoms()
print(ase_atoms)
------------------

----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:00]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:01]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:02]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:03]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:04]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:05]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:06]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:07]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:08]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:09]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:10]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:11]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:12]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:13]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:14]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:15]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:16]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:17]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:18]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:19]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:20]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:21]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:22]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:23]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:24]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:25]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:26]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:27]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:28]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:29]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:30]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:31]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:32]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:33]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:34]
----- stderr -----
Finished relaxations:   0%|          | 0/54 [00:35]
----- stderr -----

----- stderr -----

------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mgaierror[0m                                  Traceback (most recent call last)
File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/urllib3/connection.py:198[0m, in [0;36mHTTPConnection._new_conn[0;34m(self)[0m
[1;32m    197[0m [38;5;28;01mtry[39;00m:
[0;32m--> 198[0m     sock [38;5;241m=[39m [43mconnection[49m[38;5;241;43m.[39;49m[43mcreate_connection[49m[43m([49m
[1;32m    199[0m [43m        [49m[43m([49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_dns_host[49m[43m,[49m[43m [49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mport[49m[43m)[49m[43m,[49m
[1;32m    200[0m [43m        [49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mtimeout[49m[43m,[49m
[1;32m    201[0m [43m        [49m[43msource_address[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43msource_address[49m[43m,[49m
[1;32m    202[0m [43m        [49m[43msocket_options[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43msocket_options[49m[43m,[49m
[1;32m    203[0m [43m    [49m[43m)[49m
[1;32m    204[0m [38;5;28;01mexcept[39;00m socket[38;5;241m.[39mgaierror [38;5;28;01mas[39;00m e:

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/urllib3/util/connection.py:60[0m, in [0;36mcreate_connection[0;34m(address, timeout, source_address, socket_options)[0m
[1;32m     58[0m     [38;5;28;01mraise[39;00m LocationParseError([38;5;124mf[39m[38;5;124m"[39m[38;5;124m'[39m[38;5;132;01m{[39;00mhost[38;5;132;01m}[39;00m[38;5;124m'[39m[38;5;124m, label empty or too long[39m[38;5;124m"[39m) [38;5;28;01mfrom[39;00m [38;5;28;01mNone[39;00m
[0;32m---> 60[0m [38;5;28;01mfor[39;00m res [38;5;129;01min[39;00m [43msocket[49m[38;5;241;43m.[39;49m[43mgetaddrinfo[49m[43m([49m[43mhost[49m[43m,[49m[43m [49m[43mport[49m[43m,[49m[43m [49m[43mfamily[49m[43m,[49m[43m [49m[43msocket[49m[38;5;241;43m.[39;49m[43mSOCK_STREAM[49m[43m)[49m:
[1;32m     61[0m     af, socktype, proto, canonname, sa [38;5;241m=[39m res

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/socket.py:962[0m, in [0;36mgetaddrinfo[0;34m(host, port, family, type, proto, flags)[0m
[1;32m    961[0m addrlist [38;5;241m=[39m []
[0;32m--> 962[0m [38;5;28;01mfor[39;00m res [38;5;129;01min[39;00m [43m_socket[49m[38;5;241;43m.[39;49m[43mgetaddrinfo[49m[43m([49m[43mhost[49m[43m,[49m[43m [49m[43mport[49m[43m,[49m[43m [49m[43mfamily[49m[43m,[49m[43m [49m[38;5;28;43mtype[39;49m[43m,[49m[43m [49m[43mproto[49m[43m,[49m[43m [49m[43mflags[49m[43m)[49m:
[1;32m    963[0m     af, socktype, proto, canonname, sa [38;5;241m=[39m res

[0;31mgaierror[0m: [Errno -5] No address associated with hostname

The above exception was the direct cause of the following exception:

[0;31mNameResolutionError[0m                       Traceback (most recent call last)
File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/urllib3/connectionpool.py:793[0m, in [0;36mHTTPConnectionPool.urlopen[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)[0m
[1;32m    792[0m [38;5;66;03m# Make the request on the HTTPConnection object[39;00m
[0;32m--> 793[0m response [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_make_request[49m[43m([49m
[1;32m    794[0m [43m    [49m[43mconn[49m[43m,[49m
[1;32m    795[0m [43m    [49m[43mmethod[49m[43m,[49m
[1;32m    796[0m [43m    [49m[43murl[49m[43m,[49m
[1;32m    797[0m [43m    [49m[43mtimeout[49m[38;5;241;43m=[39;49m[43mtimeout_obj[49m[43m,[49m
[1;32m    798[0m [43m    [49m[43mbody[49m[38;5;241;43m=[39;49m[43mbody[49m[43m,[49m
[1;32m    799[0m [43m    [49m[43mheaders[49m[38;5;241;43m=[39;49m[43mheaders[49m[43m,[49m
[1;32m    800[0m [43m    [49m[43mchunked[49m[38;5;241;43m=[39;49m[43mchunked[49m[43m,[49m
[1;32m    801[0m [43m    [49m[43mretries[49m[38;5;241;43m=[39;49m[43mretries[49m[43m,[49m
[1;32m    802[0m [43m    [49m[43mresponse_conn[49m[38;5;241;43m=[39;49m[43mresponse_conn[49m[43m,[49m
[1;32m    803[0m [43m    [49m[43mpreload_content[49m[38;5;241;43m=[39;49m[43mpreload_content[49m[43m,[49m
[1;32m    804[0m [43m    [49m[43mdecode_content[49m[38;5;241;43m=[39;49m[43mdecode_content[49m[43m,[49m
[1;32m    805[0m [43m    [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mresponse_kw[49m[43m,[49m
[1;32m    806[0m [43m[49m[43m)[49m
[1;32m    808[0m [38;5;66;03m# Everything went great![39;00m

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/urllib3/connectionpool.py:491[0m, in [0;36mHTTPConnectionPool._make_request[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)[0m
[1;32m    490[0m         new_e [38;5;241m=[39m _wrap_proxy_error(new_e, conn[38;5;241m.[39mproxy[38;5;241m.[39mscheme)
[0;32m--> 491[0m     [38;5;28;01mraise[39;00m new_e
[1;32m    493[0m [38;5;66;03m# conn.request() calls http.client.*.request, not the method in[39;00m
[1;32m    494[0m [38;5;66;03m# urllib3.request. It also calls makefile (recv) on the socket.[39;00m

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/urllib3/connectionpool.py:467[0m, in [0;36mHTTPConnectionPool._make_request[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)[0m
[1;32m    466[0m [38;5;28;01mtry[39;00m:
[0;32m--> 467[0m     [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_validate_conn[49m[43m([49m[43mconn[49m[43m)[49m
[1;32m    468[0m [38;5;28;01mexcept[39;00m (SocketTimeout, BaseSSLError) [38;5;28;01mas[39;00m e:

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/urllib3/connectionpool.py:1099[0m, in [0;36mHTTPSConnectionPool._validate_conn[0;34m(self, conn)[0m
[1;32m   1098[0m [38;5;28;01mif[39;00m conn[38;5;241m.[39mis_closed:
[0;32m-> 1099[0m     [43mconn[49m[38;5;241;43m.[39;49m[43mconnect[49m[43m([49m[43m)[49m
[1;32m   1101[0m [38;5;66;03m# TODO revise this, see https://github.com/urllib3/urllib3/issues/2791[39;00m

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/urllib3/connection.py:616[0m, in [0;36mHTTPSConnection.connect[0;34m(self)[0m
[1;32m    615[0m sock: socket[38;5;241m.[39msocket [38;5;241m|[39m ssl[38;5;241m.[39mSSLSocket
[0;32m--> 616[0m [38;5;28mself[39m[38;5;241m.[39msock [38;5;241m=[39m sock [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_new_conn[49m[43m([49m[43m)[49m
[1;32m    617[0m server_hostname: [38;5;28mstr[39m [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mhost

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/urllib3/connection.py:205[0m, in [0;36mHTTPConnection._new_conn[0;34m(self)[0m
[1;32m    204[0m [38;5;28;01mexcept[39;00m socket[38;5;241m.[39mgaierror [38;5;28;01mas[39;00m e:
[0;32m--> 205[0m     [38;5;28;01mraise[39;00m NameResolutionError([38;5;28mself[39m[38;5;241m.[39mhost, [38;5;28mself[39m, e) [38;5;28;01mfrom[39;00m [38;5;21;01me[39;00m
[1;32m    206[0m [38;5;28;01mexcept[39;00m SocketTimeout [38;5;28;01mas[39;00m e:

[0;31mNameResolutionError[0m: <urllib3.connection.HTTPSConnection object at 0x7f7a181e2b90>: Failed to resolve 'open-catalyst-api.metademolab.com' ([Errno -5] No address associated with hostname)

The above exception was the direct cause of the following exception:

[0;31mMaxRetryError[0m                             Traceback (most recent call last)
File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/requests/adapters.py:486[0m, in [0;36mHTTPAdapter.send[0;34m(self, request, stream, timeout, verify, cert, proxies)[0m
[1;32m    485[0m [38;5;28;01mtry[39;00m:
[0;32m--> 486[0m     resp [38;5;241m=[39m [43mconn[49m[38;5;241;43m.[39;49m[43murlopen[49m[43m([49m
[1;32m    487[0m [43m        [49m[43mmethod[49m[38;5;241;43m=[39;49m[43mrequest[49m[38;5;241;43m.[39;49m[43mmethod[49m[43m,[49m
[1;32m    488[0m [43m        [49m[43murl[49m[38;5;241;43m=[39;49m[43murl[49m[43m,[49m
[1;32m    489[0m [43m        [49m[43mbody[49m[38;5;241;43m=[39;49m[43mrequest[49m[38;5;241;43m.[39;49m[43mbody[49m[43m,[49m
[1;32m    490[0m [43m        [49m[43mheaders[49m[38;5;241;43m=[39;49m[43mrequest[49m[38;5;241;43m.[39;49m[43mheaders[49m[43m,[49m
[1;32m    491[0m [43m        [49m[43mredirect[49m[38;5;241;43m=[39;49m[38;5;28;43;01mFalse[39;49;00m[43m,[49m
[1;32m    492[0m [43m        [49m[43massert_same_host[49m[38;5;241;43m=[39;49m[38;5;28;43;01mFalse[39;49;00m[43m,[49m
[1;32m    493[0m [43m        [49m[43mpreload_content[49m[38;5;241;43m=[39;49m[38;5;28;43;01mFalse[39;49;00m[43m,[49m
[1;32m    494[0m [43m        [49m[43mdecode_content[49m[38;5;241;43m=[39;49m[38;5;28;43;01mFalse[39;49;00m[43m,[49m
[1;32m    495[0m [43m        [49m[43mretries[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mmax_retries[49m[43m,[49m
[1;32m    496[0m [43m        [49m[43mtimeout[49m[38;5;241;43m=[39;49m[43mtimeout[49m[43m,[49m
[1;32m    497[0m [43m        [49m[43mchunked[49m[38;5;241;43m=[39;49m[43mchunked[49m[43m,[49m
[1;32m    498[0m [43m    [49m[43m)[49m
[1;32m    500[0m [38;5;28;01mexcept[39;00m (ProtocolError, [38;5;167;01mOSError[39;00m) [38;5;28;01mas[39;00m err:

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/urllib3/connectionpool.py:847[0m, in [0;36mHTTPConnectionPool.urlopen[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)[0m
[1;32m    845[0m     new_e [38;5;241m=[39m ProtocolError([38;5;124m"[39m[38;5;124mConnection aborted.[39m[38;5;124m"[39m, new_e)
[0;32m--> 847[0m retries [38;5;241m=[39m [43mretries[49m[38;5;241;43m.[39;49m[43mincrement[49m[43m([49m
[1;32m    848[0m [43m    [49m[43mmethod[49m[43m,[49m[43m [49m[43murl[49m[43m,[49m[43m [49m[43merror[49m[38;5;241;43m=[39;49m[43mnew_e[49m[43m,[49m[43m [49m[43m_pool[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[43m,[49m[43m [49m[43m_stacktrace[49m[38;5;241;43m=[39;49m[43msys[49m[38;5;241;43m.[39;49m[43mexc_info[49m[43m([49m[43m)[49m[43m[[49m[38;5;241;43m2[39;49m[43m][49m
[1;32m    849[0m [43m[49m[43m)[49m
[1;32m    850[0m retries[38;5;241m.[39msleep()

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/urllib3/util/retry.py:515[0m, in [0;36mRetry.increment[0;34m(self, method, url, response, error, _pool, _stacktrace)[0m
[1;32m    514[0m     reason [38;5;241m=[39m error [38;5;129;01mor[39;00m ResponseError(cause)
[0;32m--> 515[0m     [38;5;28;01mraise[39;00m MaxRetryError(_pool, url, reason) [38;5;28;01mfrom[39;00m [38;5;21;01mreason[39;00m  [38;5;66;03m# type: ignore[arg-type][39;00m
[1;32m    517[0m log[38;5;241m.[39mdebug([38;5;124m"[39m[38;5;124mIncremented Retry for (url=[39m[38;5;124m'[39m[38;5;132;01m%s[39;00m[38;5;124m'[39m[38;5;124m): [39m[38;5;132;01m%r[39;00m[38;5;124m"[39m, url, new_retry)

[0;31mMaxRetryError[0m: HTTPSConnectionPool(host='open-catalyst-api.metademolab.com', port=443): Max retries exceeded with url: /ocp/adsorbate-slab-relaxations/405820fc-babf-48ed-b51b-062882235bc2/configs?field=energy (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x7f7a181e2b90>: Failed to resolve 'open-catalyst-api.metademolab.com' ([Errno -5] No address associated with hostname)"))

During handling of the above exception, another exception occurred:

[0;31mConnectionError[0m                           Traceback (most recent call last)
File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ocpapi/client/client.py:401[0m, in [0;36mClient._run_request[0;34m(self, path, method, **kwargs)[0m
[1;32m    400[0m [38;5;28;01mtry[39;00m:
[0;32m--> 401[0m     response: requests[38;5;241m.[39mResponse [38;5;241m=[39m [38;5;28;01mawait[39;00m asyncio[38;5;241m.[39mto_thread(
[1;32m    402[0m         requests[38;5;241m.[39mrequest,
[1;32m    403[0m         method[38;5;241m=[39mmethod,
[1;32m    404[0m         url[38;5;241m=[39murl,
[1;32m    405[0m         [38;5;241m*[39m[38;5;241m*[39mkwargs,
[1;32m    406[0m     )
[1;32m    407[0m [38;5;28;01mexcept[39;00m [38;5;167;01mException[39;00m [38;5;28;01mas[39;00m e:

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/asyncio/threads.py:25[0m, in [0;36mto_thread[0;34m(func, *args, **kwargs)[0m
[1;32m     24[0m func_call [38;5;241m=[39m functools[38;5;241m.[39mpartial(ctx[38;5;241m.[39mrun, func, [38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs)
[0;32m---> 25[0m [38;5;28;01mreturn[39;00m [38;5;28;01mawait[39;00m loop[38;5;241m.[39mrun_in_executor([38;5;28;01mNone[39;00m, func_call)

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/concurrent/futures/thread.py:58[0m, in [0;36m_WorkItem.run[0;34m(self)[0m
[1;32m     57[0m [38;5;28;01mtry[39;00m:
[0;32m---> 58[0m     result [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mfn[49m[43m([49m[38;5;241;43m*[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43margs[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mkwargs[49m[43m)[49m
[1;32m     59[0m [38;5;28;01mexcept[39;00m [38;5;167;01mBaseException[39;00m [38;5;28;01mas[39;00m exc:

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/requests/api.py:59[0m, in [0;36mrequest[0;34m(method, url, **kwargs)[0m
[1;32m     58[0m [38;5;28;01mwith[39;00m sessions[38;5;241m.[39mSession() [38;5;28;01mas[39;00m session:
[0;32m---> 59[0m     [38;5;28;01mreturn[39;00m [43msession[49m[38;5;241;43m.[39;49m[43mrequest[49m[43m([49m[43mmethod[49m[38;5;241;43m=[39;49m[43mmethod[49m[43m,[49m[43m [49m[43murl[49m[38;5;241;43m=[39;49m[43murl[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/requests/sessions.py:589[0m, in [0;36mSession.request[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)[0m
[1;32m    588[0m send_kwargs[38;5;241m.[39mupdate(settings)
[0;32m--> 589[0m resp [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43msend[49m[43m([49m[43mprep[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43msend_kwargs[49m[43m)[49m
[1;32m    591[0m [38;5;28;01mreturn[39;00m resp

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/requests/sessions.py:703[0m, in [0;36mSession.send[0;34m(self, request, **kwargs)[0m
[1;32m    702[0m [38;5;66;03m# Send the request[39;00m
[0;32m--> 703[0m r [38;5;241m=[39m [43madapter[49m[38;5;241;43m.[39;49m[43msend[49m[43m([49m[43mrequest[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m
[1;32m    705[0m [38;5;66;03m# Total elapsed time of the request (approximately)[39;00m

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/requests/adapters.py:519[0m, in [0;36mHTTPAdapter.send[0;34m(self, request, stream, timeout, verify, cert, proxies)[0m
[1;32m    517[0m         [38;5;28;01mraise[39;00m SSLError(e, request[38;5;241m=[39mrequest)
[0;32m--> 519[0m     [38;5;28;01mraise[39;00m [38;5;167;01mConnectionError[39;00m(e, request[38;5;241m=[39mrequest)
[1;32m    521[0m [38;5;28;01mexcept[39;00m ClosedPoolError [38;5;28;01mas[39;00m e:

[0;31mConnectionError[0m: HTTPSConnectionPool(host='open-catalyst-api.metademolab.com', port=443): Max retries exceeded with url: /ocp/adsorbate-slab-relaxations/405820fc-babf-48ed-b51b-062882235bc2/configs?field=energy (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x7f7a181e2b90>: Failed to resolve 'open-catalyst-api.metademolab.com' ([Errno -5] No address associated with hostname)"))

The above exception was the direct cause of the following exception:

[0;31mRequestException[0m                          Traceback (most recent call last)
Cell [0;32mIn[8], line 3[0m
[1;32m      1[0m [38;5;28;01mfrom[39;00m [38;5;21;01mocpapi[39;00m [38;5;28;01mimport[39;00m find_adsorbate_binding_sites
[0;32m----> 3[0m results [38;5;241m=[39m [38;5;28;01mawait[39;00m find_adsorbate_binding_sites(
[1;32m      4[0m     adsorbate[38;5;241m=[39m[38;5;124m"[39m[38;5;124m*OH[39m[38;5;124m"[39m,
[1;32m      5[0m     bulk[38;5;241m=[39m[38;5;124m"[39m[38;5;124mmp-126[39m[38;5;124m"[39m,
[1;32m      6[0m     adslab_filter[38;5;241m=[39mkeep_slabs_with_miller_indices([([38;5;241m1[39m, [38;5;241m1[39m, [38;5;241m1[39m)]),
[1;32m      7[0m )
[1;32m      9[0m ase_atoms [38;5;241m=[39m results[38;5;241m.[39mslabs[[38;5;241m0[39m][38;5;241m.[39mconfigs[[38;5;241m0[39m][38;5;241m.[39mto_ase_atoms()
[1;32m     10[0m [38;5;28mprint[39m(ase_atoms)

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ocpapi/workflows/adsorbates.py:908[0m, in [0;36mfind_adsorbate_binding_sites[0;34m(adsorbate, bulk, model, adslab_filter, client, lifetime)[0m
[1;32m    905[0m adslabs [38;5;241m=[39m [38;5;28;01mawait[39;00m adslab_filter(adslabs)
[1;32m    907[0m [38;5;66;03m# Find binding sites on all slabs[39;00m
[0;32m--> 908[0m [38;5;28;01mreturn[39;00m [38;5;28;01mawait[39;00m _relax_binding_sites_on_slabs(
[1;32m    909[0m     client[38;5;241m=[39mclient,
[1;32m    910[0m     adsorbate[38;5;241m=[39madsorbate,
[1;32m    911[0m     bulk[38;5;241m=[39mbulk_obj,
[1;32m    912[0m     adslabs[38;5;241m=[39madslabs,
[1;32m    913[0m     model[38;5;241m=[39mmodel,
[1;32m    914[0m     lifetime[38;5;241m=[39mlifetime,
[1;32m    915[0m )

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ocpapi/workflows/adsorbates.py:811[0m, in [0;36m_relax_binding_sites_on_slabs[0;34m(client, adsorbate, bulk, adslabs, model, lifetime)[0m
[1;32m    804[0m             [38;5;28;01mawait[39;00m pbar_refresh_task
[1;32m    806[0m [38;5;66;03m# Return results[39;00m
[1;32m    807[0m [38;5;28;01mreturn[39;00m AdsorbateBindingSites(
[1;32m    808[0m     adsorbate[38;5;241m=[39madsorbate,
[1;32m    809[0m     bulk[38;5;241m=[39mbulk,
[1;32m    810[0m     model[38;5;241m=[39mmodel,
[0;32m--> 811[0m     slabs[38;5;241m=[39m[43m[[49m[43mt[49m[38;5;241;43m.[39;49m[43mresult[49m[43m([49m[43m)[49m[43m [49m[38;5;28;43;01mfor[39;49;00m[43m [49m[43mt[49m[43m [49m[38;5;129;43;01min[39;49;00m[43m [49m[43mtasks[49m[43m][49m,
[1;32m    812[0m )

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ocpapi/workflows/adsorbates.py:811[0m, in [0;36m<listcomp>[0;34m(.0)[0m
[1;32m    804[0m             [38;5;28;01mawait[39;00m pbar_refresh_task
[1;32m    806[0m [38;5;66;03m# Return results[39;00m
[1;32m    807[0m [38;5;28;01mreturn[39;00m AdsorbateBindingSites(
[1;32m    808[0m     adsorbate[38;5;241m=[39madsorbate,
[1;32m    809[0m     bulk[38;5;241m=[39mbulk,
[1;32m    810[0m     model[38;5;241m=[39mmodel,
[0;32m--> 811[0m     slabs[38;5;241m=[39m[[43mt[49m[38;5;241;43m.[39;49m[43mresult[49m[43m([49m[43m)[49m [38;5;28;01mfor[39;00m t [38;5;129;01min[39;00m tasks],
[1;32m    812[0m )

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ocpapi/workflows/adsorbates.py:699[0m, in [0;36m_run_relaxations_on_slab[0;34m(client, adsorbate, adsorbate_configs, bulk, slab, model, lifetime, pbar)[0m
[1;32m    694[0m     [38;5;28;01mawait[39;00m es[38;5;241m.[39menter_async_context(
[1;32m    695[0m         _ensure_system_deleted(client[38;5;241m=[39mclient, system_id[38;5;241m=[39msystem_id)
[1;32m    696[0m     )
[1;32m    698[0m [38;5;66;03m# Wait for all relaxations to finish[39;00m
[0;32m--> 699[0m [38;5;28;01mawait[39;00m wait_for_adsorbate_slab_relaxations(
[1;32m    700[0m     system_id[38;5;241m=[39msystem_id,
[1;32m    701[0m     pbar[38;5;241m=[39mpbar,
[1;32m    702[0m     client[38;5;241m=[39mclient,
[1;32m    703[0m )
[1;32m    705[0m [38;5;66;03m# Fetch the final results[39;00m
[1;32m    706[0m results: List[
[1;32m    707[0m     AdsorbateSlabRelaxationResult
[1;32m    708[0m ] [38;5;241m=[39m [38;5;28;01mawait[39;00m get_adsorbate_slab_relaxation_results(
[1;32m    709[0m     client[38;5;241m=[39mclient,
[1;32m    710[0m     system_id[38;5;241m=[39msystem_id,
[1;32m    711[0m )

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ocpapi/workflows/adsorbates.py:584[0m, in [0;36mwait_for_adsorbate_slab_relaxations[0;34m(system_id, check_immediately, slow_interval_sec, fast_interval_sec, pbar, client)[0m
[1;32m    578[0m num_finished: [38;5;28mint[39m [38;5;241m=[39m [38;5;241m0[39m
[1;32m    579[0m [38;5;28;01mwhile[39;00m [38;5;28;01mTrue[39;00m:
[1;32m    580[0m     [38;5;66;03m# Get the current results. Only fetch the energy; this hits an index[39;00m
[1;32m    581[0m     [38;5;66;03m# that will return results more quickly.[39;00m
[1;32m    582[0m     results: List[
[1;32m    583[0m         AdsorbateSlabRelaxationResult
[0;32m--> 584[0m     ] [38;5;241m=[39m [38;5;28;01mawait[39;00m get_adsorbate_slab_relaxation_results(
[1;32m    585[0m         client[38;5;241m=[39mclient,
[1;32m    586[0m         system_id[38;5;241m=[39msystem_id,
[1;32m    587[0m         fields[38;5;241m=[39m[[38;5;124m"[39m[38;5;124menergy[39m[38;5;124m"[39m],
[1;32m    588[0m     )
[1;32m    590[0m     [38;5;66;03m# Check the number of finished jobs[39;00m
[1;32m    591[0m     last_num_finished: [38;5;28mint[39m [38;5;241m=[39m num_finished

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/tenacity/_asyncio.py:88[0m, in [0;36mAsyncRetrying.wraps.<locals>.async_wrapped[0;34m(*args, **kwargs)[0m
[1;32m     86[0m [38;5;129m@functools[39m[38;5;241m.[39mwraps(fn)
[1;32m     87[0m [38;5;28;01masync[39;00m [38;5;28;01mdef[39;00m [38;5;21masync_wrapped[39m([38;5;241m*[39margs: t[38;5;241m.[39mAny, [38;5;241m*[39m[38;5;241m*[39mkwargs: t[38;5;241m.[39mAny) [38;5;241m-[39m[38;5;241m>[39m t[38;5;241m.[39mAny:
[0;32m---> 88[0m     [38;5;28;01mreturn[39;00m [38;5;28;01mawait[39;00m fn([38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs)

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/tenacity/_asyncio.py:47[0m, in [0;36mAsyncRetrying.__call__[0;34m(self, fn, *args, **kwargs)[0m
[1;32m     45[0m retry_state [38;5;241m=[39m RetryCallState(retry_object[38;5;241m=[39m[38;5;28mself[39m, fn[38;5;241m=[39mfn, args[38;5;241m=[39margs, kwargs[38;5;241m=[39mkwargs)
[1;32m     46[0m [38;5;28;01mwhile[39;00m [38;5;28;01mTrue[39;00m:
[0;32m---> 47[0m     do [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43miter[49m[43m([49m[43mretry_state[49m[38;5;241;43m=[39;49m[43mretry_state[49m[43m)[49m
[1;32m     48[0m     [38;5;28;01mif[39;00m [38;5;28misinstance[39m(do, DoAttempt):
[1;32m     49[0m         [38;5;28;01mtry[39;00m:

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/tenacity/__init__.py:325[0m, in [0;36mBaseRetrying.iter[0;34m(self, retry_state)[0m
[1;32m    323[0m     retry_exc [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mretry_error_cls(fut)
[1;32m    324[0m     [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39mreraise:
[0;32m--> 325[0m         [38;5;28;01mraise[39;00m [43mretry_exc[49m[38;5;241;43m.[39;49m[43mreraise[49m[43m([49m[43m)[49m
[1;32m    326[0m     [38;5;28;01mraise[39;00m retry_exc [38;5;28;01mfrom[39;00m [38;5;21;01mfut[39;00m[38;5;21;01m.[39;00m[38;5;21;01mexception[39;00m()
[1;32m    328[0m [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39mwait:

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/tenacity/__init__.py:158[0m, in [0;36mRetryError.reraise[0;34m(self)[0m
[1;32m    156[0m [38;5;28;01mdef[39;00m [38;5;21mreraise[39m([38;5;28mself[39m) [38;5;241m-[39m[38;5;241m>[39m t[38;5;241m.[39mNoReturn:
[1;32m    157[0m     [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39mlast_attempt[38;5;241m.[39mfailed:
[0;32m--> 158[0m         [38;5;28;01mraise[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mlast_attempt[49m[38;5;241;43m.[39;49m[43mresult[49m[43m([49m[43m)[49m
[1;32m    159[0m     [38;5;28;01mraise[39;00m [38;5;28mself[39m

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/concurrent/futures/_base.py:449[0m, in [0;36mFuture.result[0;34m(self, timeout)[0m
[1;32m    447[0m     [38;5;28;01mraise[39;00m CancelledError()
[1;32m    448[0m [38;5;28;01melif[39;00m [38;5;28mself[39m[38;5;241m.[39m_state [38;5;241m==[39m FINISHED:
[0;32m--> 449[0m     [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m__get_result[49m[43m([49m[43m)[49m
[1;32m    451[0m [38;5;28mself[39m[38;5;241m.[39m_condition[38;5;241m.[39mwait(timeout)
[1;32m    453[0m [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39m_state [38;5;129;01min[39;00m [CANCELLED, CANCELLED_AND_NOTIFIED]:

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/concurrent/futures/_base.py:401[0m, in [0;36mFuture.__get_result[0;34m(self)[0m
[1;32m    399[0m [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39m_exception:
[1;32m    400[0m     [38;5;28;01mtry[39;00m:
[0;32m--> 401[0m         [38;5;28;01mraise[39;00m [38;5;28mself[39m[38;5;241m.[39m_exception
[1;32m    402[0m     [38;5;28;01mfinally[39;00m:
[1;32m    403[0m         [38;5;66;03m# Break a reference cycle with the exception in self._exception[39;00m
[1;32m    404[0m         [38;5;28mself[39m [38;5;241m=[39m [38;5;28;01mNone[39;00m

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/tenacity/_asyncio.py:50[0m, in [0;36mAsyncRetrying.__call__[0;34m(self, fn, *args, **kwargs)[0m
[1;32m     48[0m [38;5;28;01mif[39;00m [38;5;28misinstance[39m(do, DoAttempt):
[1;32m     49[0m     [38;5;28;01mtry[39;00m:
[0;32m---> 50[0m         result [38;5;241m=[39m [38;5;28;01mawait[39;00m fn([38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs)
[1;32m     51[0m     [38;5;28;01mexcept[39;00m [38;5;167;01mBaseException[39;00m:  [38;5;66;03m# noqa: B902[39;00m
[1;32m     52[0m         retry_state[38;5;241m.[39mset_exception(sys[38;5;241m.[39mexc_info())  [38;5;66;03m# type: ignore[arg-type][39;00m

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ocpapi/workflows/adsorbates.py:512[0m, in [0;36mget_adsorbate_slab_relaxation_results[0;34m(system_id, config_ids, fields, client)[0m
[1;32m    487[0m [38;5;129m@retry_api_calls[39m(max_attempts[38;5;241m=[39m[38;5;241m3[39m)
[1;32m    488[0m [38;5;28;01masync[39;00m [38;5;28;01mdef[39;00m [38;5;21mget_adsorbate_slab_relaxation_results[39m(
[1;32m    489[0m     system_id: [38;5;28mstr[39m,
[0;32m   (...)[0m
[1;32m    492[0m     client: Client [38;5;241m=[39m DEFAULT_CLIENT,
[1;32m    493[0m ) [38;5;241m-[39m[38;5;241m>[39m List[AdsorbateSlabRelaxationResult]:
[1;32m    494[0m [38;5;250m    [39m[38;5;124;03m"""[39;00m
[1;32m    495[0m [38;5;124;03m    Wrapper around Client.get_adsorbate_slab_relaxations_results() that[39;00m
[1;32m    496[0m [38;5;124;03m    handles retries, including re-fetching individual configurations that[39;00m
[0;32m   (...)[0m
[1;32m    509[0m [38;5;124;03m        the system.[39;00m
[1;32m    510[0m [38;5;124;03m    """[39;00m
[1;32m    511[0m     results: AdsorbateSlabRelaxationsResults [38;5;241m=[39m (
[0;32m--> 512[0m         [38;5;28;01mawait[39;00m client[38;5;241m.[39mget_adsorbate_slab_relaxations_results(
[1;32m    513[0m             system_id[38;5;241m=[39msystem_id,
[1;32m    514[0m             config_ids[38;5;241m=[39mconfig_ids,
[1;32m    515[0m             fields[38;5;241m=[39mfields,
[1;32m    516[0m         )
[1;32m    517[0m     )
[1;32m    519[0m     [38;5;66;03m# Save a copy of all results that were fetched[39;00m
[1;32m    520[0m     fetched: List[AdsorbateSlabRelaxationResult] [38;5;241m=[39m [38;5;28mlist[39m(results[38;5;241m.[39mconfigs)

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ocpapi/client/client.py:350[0m, in [0;36mClient.get_adsorbate_slab_relaxations_results[0;34m(self, system_id, config_ids, fields)[0m
[1;32m    348[0m [38;5;28;01mif[39;00m config_ids:
[1;32m    349[0m     params[[38;5;124m"[39m[38;5;124mconfig_id[39m[38;5;124m"[39m] [38;5;241m=[39m config_ids
[0;32m--> 350[0m response: [38;5;28mstr[39m [38;5;241m=[39m [38;5;28;01mawait[39;00m [38;5;28mself[39m[38;5;241m.[39m_run_request(
[1;32m    351[0m     path[38;5;241m=[39m[38;5;124mf[39m[38;5;124m"[39m[38;5;124mocp/adsorbate-slab-relaxations/[39m[38;5;132;01m{[39;00msystem_id[38;5;132;01m}[39;00m[38;5;124m/configs[39m[38;5;124m"[39m,
[1;32m    352[0m     method[38;5;241m=[39m[38;5;124m"[39m[38;5;124mGET[39m[38;5;124m"[39m,
[1;32m    353[0m     params[38;5;241m=[39mparams,
[1;32m    354[0m )
[1;32m    355[0m [38;5;28;01mreturn[39;00m AdsorbateSlabRelaxationsResults[38;5;241m.[39mfrom_json(response)

File [0;32m/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ocpapi/client/client.py:408[0m, in [0;36mClient._run_request[0;34m(self, path, method, **kwargs)[0m
[1;32m    401[0m     response: requests[38;5;241m.[39mResponse [38;5;241m=[39m [38;5;28;01mawait[39;00m asyncio[38;5;241m.[39mto_thread(
[1;32m    402[0m         requests[38;5;241m.[39mrequest,
[1;32m    403[0m         method[38;5;241m=[39mmethod,
[1;32m    404[0m         url[38;5;241m=[39murl,
[1;32m    405[0m         [38;5;241m*[39m[38;5;241m*[39mkwargs,
[1;32m    406[0m     )
[1;32m    407[0m [38;5;28;01mexcept[39;00m [38;5;167;01mException[39;00m [38;5;28;01mas[39;00m e:
[0;32m--> 408[0m     [38;5;28;01mraise[39;00m RequestException(
[1;32m    409[0m         method[38;5;241m=[39mmethod,
[1;32m    410[0m         url[38;5;241m=[39murl,
[1;32m    411[0m         cause[38;5;241m=[39m[38;5;124mf[39m[38;5;124m"[39m[38;5;124mException while making request: [39m[38;5;132;01m{[39;00m[38;5;28mtype[39m(e)[38;5;241m.[39m[38;5;18m__name__[39m[38;5;132;01m}[39;00m[38;5;124m: [39m[38;5;132;01m{[39;00me[38;5;132;01m}[39;00m[38;5;124m"[39m,
[1;32m    412[0m     ) [38;5;28;01mfrom[39;00m [38;5;21;01me[39;00m
[1;32m    414[0m [38;5;66;03m# Check the response code[39;00m
[1;32m    415[0m [38;5;28;01mif[39;00m response[38;5;241m.[39mstatus_code [38;5;241m>[39m[38;5;241m=[39m [38;5;241m300[39m:
[1;32m    416[0m     [38;5;66;03m# Exceeded server side rate limit[39;00m

[0;31mRequestException[0m: Request to GET https://open-catalyst-api.metademolab.com/ocp/adsorbate-slab-relaxations/405820fc-babf-48ed-b51b-062882235bc2/configs failed. Exception while making request: ConnectionError: HTTPSConnectionPool(host='open-catalyst-api.metademolab.com', port=443): Max retries exceeded with url: /ocp/adsorbate-slab-relaxations/405820fc-babf-48ed-b51b-062882235bc2/configs?field=energy (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x7f7a181e2b90>: Failed to resolve 'open-catalyst-api.metademolab.com' ([Errno -5] No address associated with hostname)"))

